packages/amplify_datastore/test/native_amplify_api_test.dart

```diff
@@ -0,0 +1,971 @@
+            List responses = message as List;
+
+            expect(responses.length, greaterThan(0));
+            expect(responses[0], isA<NativeGraphQLSubscriptionResponse>());
+            NativeGraphQLSubscriptionResponse response = responses[0];
+
+            expect(response.payloadJson, errorsJson);
+            expect(response.subscriptionId.length, greaterThan(0));
+            expect(response.type, 'error');
+
+            return Future.value(message);
+          },
+        );
+
+        expect(responseController, isNotNull);
+        expect(
+          () => responseController?.addError(Exception(exceptionMessage)),
+          returnsNormally,
+        );
+      });
+
+      test('Send Done Event', () async {
+        StreamController? responseController;
+
+        mockAPIPlugin.subscribeMethod = <String>(
+          GraphQLRequest mockRequest,
+          void Function()? onEstablished,
+        ) {
+          var controller = StreamController<GraphQLResponse<String>>();
+          responseController = controller;
+          return controller.stream;
+        };
+
+        NativeAmplifyApi nativeAmplifyApi = NativeAmplifyApi({});
+
+        await nativeAmplifyApi.subscribe(
+          NativeGraphQLRequest(document: ''),
+        );
+
+        binding.defaultBinaryMessenger.setMockDecodedMessageHandler<Object?>(
+          channel,
+          (Object? message) {
+            expect(message, isA<List>());
+            List responses = message as List;
+
+            expect(responses.length, greaterThan(0));
+            expect(responses[0], isA<NativeGraphQLSubscriptionResponse>());
+            NativeGraphQLSubscriptionResponse response = responses[0];
+
+            expect(response.payloadJson, null);
+            expect(response.subscriptionId.length, greaterThan(0));
+            expect(response.type, 'complete');
+
+            return Future.value(message);
+          },
+        );
+
+        expect(responseController, isNotNull);
+        expect(
+          () => responseController?.close(),
+          returnsNormally,
+        );
+      });
+
+      group('Unubscribe', () {
+        test('Existing Subscription', () async {
+          mockAPIPlugin.subscribeMethod = <String>(
+            GraphQLRequest mockRequest,
+            void Function()? onEstablished,
+          ) {
+            var controller = StreamController<GraphQLResponse<String>>();
+            return controller.stream;
+          };
+
+          NativeAmplifyApi nativeAmplifyApi = NativeAmplifyApi({});
+
+          var response = await nativeAmplifyApi.subscribe(
+            NativeGraphQLRequest(
+              document: '',
+            ),
+          );
+
+          await nativeAmplifyApi.unsubscribe(response.subscriptionId);
+        });
+
+        test('Fake Subscription', () async {
+          NativeAmplifyApi nativeAmplifyApi = NativeAmplifyApi({});
+
+          await nativeAmplifyApi.unsubscribe('Fake Subscription ID');
```

Comment:
It's a test to make sure we don't throw an exception, which would fail the unit test. To make this more explicit I added an expect returnsNormally check.